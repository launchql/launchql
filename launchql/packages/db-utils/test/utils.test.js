process.env.SKITCH_PATH = __dirname + '/../__fixtures__/skitch';

import {
  listModules,
  latestChange,
  getPlan,
  getExtensionsAndModules,
  getExtensionsAndModulesChanges
} from '../src/index';

const cleanText = (t) =>
  t
    .split('\n')
    .map((a) => a.trim())
    .filter((a) => a)
    .join('\n');

describe('sqitch modules', () => {
  it('should get modules', async () => {
    const modules = await listModules();
    expect(modules).toMatchSnapshot();
  });
  it('should get a modules last path', async () => {
    const change = await latestChange('totp');
    expect(change).toEqual('procedures/generate_secret');
  });
  it('should be able to create a deps for cross-project requires', async () => {
    const deps = await getExtensionsAndModules('utils');
    expect(deps).toEqual({
      native: ['plpgsql', 'uuid-ossp'],
      sqitch: ['totp', 'pg-verify']
    });
  });
  it('should be able to create a deps for cross-project requires with changes', async () => {
    const deps = await getExtensionsAndModulesChanges('utils');
    expect(deps).toMatchSnapshot();
  });
});
