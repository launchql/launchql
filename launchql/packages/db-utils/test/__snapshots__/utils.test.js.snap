// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqitch modules should be able to create a deps for cross-project requires with changes 1`] = `
Object {
  "native": Array [
    "plpgsql",
    "uuid-ossp",
  ],
  "sqitch": Array [
    Object {
      "latest": "procedures/generate_secret",
      "name": "totp",
      "version": "0.0.1",
    },
    Object {
      "latest": "procedures/verify_view",
      "name": "pg-verify",
      "version": "0.0.1",
    },
  ],
}
`;

exports[`sqitch modules should get modules 1`] = `
Object {
  "pg-utilities": Object {
    "path": "packages/utilities",
    "requires": Array [
      "plpgsql",
    ],
    "version": "0.0.1",
  },
  "pg-verify": Object {
    "path": "packages/verify",
    "requires": Array [
      "plpgsql",
      "uuid-ossp",
      "pg-utilities",
    ],
    "version": "0.0.1",
  },
  "secrets": Object {
    "path": "packages/secrets",
    "requires": Array [
      "plpgsql",
      "uuid-ossp",
      "totp",
      "pg-verify",
    ],
    "version": "0.0.1",
  },
  "totp": Object {
    "path": "packages/totp",
    "requires": Array [
      "plpgsql",
      "uuid-ossp",
      "pgcrypto",
      "pg-verify",
    ],
    "version": "0.0.1",
  },
  "utils": Object {
    "path": "packages/utils",
    "requires": Array [
      "plpgsql",
      "uuid-ossp",
      "totp",
      "pg-verify",
    ],
    "version": "0.0.1",
  },
}
`;
